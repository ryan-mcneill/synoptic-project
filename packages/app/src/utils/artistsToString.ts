import { BasicDetails } from "../store/types";

export const artistsToString = (artists: Partial<BasicDetails>[]): string =>
  artists
    .reduce((acc, { name }) => {
      // @ts-ignore
      acc.push(name);
      return acc;
    }, [])
    .join(", ");
