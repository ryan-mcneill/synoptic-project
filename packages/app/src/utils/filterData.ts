import { ArtistsAndAlbumsData } from "../store/types";

const filterData = (
  data: ArtistsAndAlbumsData | undefined,
  searchFor: string
) => {
  let filteredData = data ? { ...data } : data;
  if (filteredData) {
    // @ts-ignore
    filteredData = Object.keys(filteredData).reduce((acc: {}, curr) => {
      if (
        filteredData &&
        filteredData[curr] &&
        filteredData[curr].name.toLowerCase().includes(searchFor.toLowerCase())
      ) {
        acc = { ...acc, [curr]: filteredData[curr] };
      }
      return acc;
    }, {});
  }
  return filteredData;
};

export { filterData };
