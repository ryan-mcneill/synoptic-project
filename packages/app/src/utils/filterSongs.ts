import { Song } from "../store/types";

const filterSongs = (data: Song[] | undefined, searchFor: string) => {
  let filteredData = data ? data : [];
  if (filteredData) {
    filteredData = filteredData.reduce((acc: Song[], curr) => {
      if (curr.name.toLowerCase().includes(searchFor.toLowerCase())) {
        acc.push(curr);
      }
      return acc;
    }, []);
  }
  return filteredData;
};

export { filterSongs };
