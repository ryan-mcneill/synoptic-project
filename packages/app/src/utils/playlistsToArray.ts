import { ArtistsAndAlbumsData } from "../store/types";

export const playlistsToArray = (playlists: ArtistsAndAlbumsData) =>
  Object.keys(playlists).map((id) => ({ id, name: playlists[id].name }));
