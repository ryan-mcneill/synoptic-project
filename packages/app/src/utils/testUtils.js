import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../store/reducers";

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
