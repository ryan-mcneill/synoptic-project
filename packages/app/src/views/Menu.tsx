import React, { FC, ReactElement } from "react";
import { Action } from "redux";
import { connect, ConnectedProps } from "react-redux";
import { ThunkDispatch } from "redux-thunk";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEllipsisH } from "@fortawesome/free-solid-svg-icons";
import { Tabs } from "../components/Menu";
import { setIsOpen } from "../store/actions";
import { State } from "../store/types";

const mapDispatchToProps = (dispatch: ThunkDispatch<State, void, Action>) => ({
  setIsOpen: (isOpen?: boolean) => dispatch(setIsOpen(isOpen))
});

const mapStateToProps = (state: State) => ({
  albumsLoading: state.albums.loading,
  artistsLoading: state.artists.loading,
  isOpen: state.menu.isOpen
});

const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

const Menu: FC<PropsFromRedux> = ({
  albumsLoading,
  artistsLoading,
  isOpen,
  setIsOpen
}): ReactElement =>
  !albumsLoading && !artistsLoading ? (
    <div
      style={{
        position: "absolute",
        bottom: isOpen ? 0 : "-90vh",
        transition: "all 0.5s ease"
      }}
    >
      <div
        style={{
          textAlign: "center",
          fontSize: "72px",
          position: "relative",
          left: "50%",
          bottom: "-1px",
          transform: "translateX(-50%)",
          height: "72px",
          width: "128px",
          backgroundColor: "rgba(255, 255, 255, .5)",
          borderRadius: "24px 24px 0 0",
          backdropFilter: "blur(20px)"
        }}
        onClick={() => setIsOpen()}
      >
        <FontAwesomeIcon icon={faEllipsisH} />
      </div>
      <Tabs />
    </div>
  ) : (
    <></>
  );

export default connector(Menu);
