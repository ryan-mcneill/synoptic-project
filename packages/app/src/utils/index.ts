import { artistsToString } from "./artistsToString";
import { filterData } from "./filterData";
import { filterSongs } from "./filterSongs";
import { playlistsToArray } from "./playlistsToArray";

export { artistsToString, filterData, filterSongs, playlistsToArray };
