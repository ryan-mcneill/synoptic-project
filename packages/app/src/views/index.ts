import Menu from "./Menu";
import MusicPlayer from "./MusicPlayer";
import PageLoader from "./PageLoader";

export { Menu, MusicPlayer, PageLoader };
