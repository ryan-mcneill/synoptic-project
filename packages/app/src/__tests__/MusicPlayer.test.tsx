import React from "react";
import { MusicPlayer } from "../views";
import { render } from "../utils/testUtils";

describe("The MusicPlayer component", () => {
  it("renders as default with dark UI elements on a light background", () => {
    const { container } = render(<MusicPlayer />, {
      initialState: { currentSong: {} }
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with dark UI elements when the background is light", () => {
    const { container } = render(<MusicPlayer />, {
      initialState: { currentSong: { background: { isDark: false } } }
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with light UI elements when the background is dark", () => {
    const { container } = render(<MusicPlayer />, {
      initialState: { currentSong: { background: { isDark: true } } }
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
