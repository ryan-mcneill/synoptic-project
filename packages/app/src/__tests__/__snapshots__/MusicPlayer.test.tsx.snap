// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MusicPlayer component renders as default with dark UI elements on a light background 1`] = `
<div
  style="display: block; margin: 0px; position: absolute; height: 100%; width: 100%; color: black; font-size: 18px;"
>
  <div
    style="height: 5vh; width: 90vw; text-align: center; padding: 5vh 5vw 2vh 5vw;"
  >
    <div
      style="font-size: 24px;"
    >
      NOTHING PLAYING
    </div>
    <div
      style="font-size: 48px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
    >
      Please make a song selection
    </div>
  </div>
  <div
    class="album-art"
    style="width: 80vw; height: 80vw; position: absolute; top: 42vh; left: 50vw; transform: translate(-50%, -50%); background-color: rgb(192, 192, 192); box-shadow: 0px 0px 64px 4px rgba(0,0,0,0.2);"
  >
    <div
      style="font-size: 400px; position: absolute; top: 50%; left: 48%; transform: translate(-50%, -50%);"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-music fa-w-16 "
        data-icon="music"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    style="height: 10vh; width: 90vw; position: absolute; bottom: 0px; padding: 0px 5vw 5vh 5vw;"
  >
    <audio
      class="react-audio-player "
      id=""
      preload="metadata"
      src=""
      title=""
    >
      <p>
        Your browser does not support the 
        <code>
          audio
        </code>
         element.
      </p>
    </audio>
    <div
      style="display: flex; align-items: center; justify-content: center;"
    >
      <div
        style="font-size: 48px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-step-backward fa-w-14 "
          data-icon="step-backward"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 468V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v176.4l195.5-181C352.1 22.3 384 36.6 384 64v384c0 27.4-31.9 41.7-52.5 24.6L136 292.7V468c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        style="font-size: 96px; margin: 0px 96px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-play fa-w-14 "
          data-icon="play"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        style="font-size: 48px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-step-forward fa-w-14 "
          data-icon="step-forward"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 44v424c0 6.6-5.4 12-12 12h-48c-6.6 0-12-5.4-12-12V291.6l-195.5 181C95.9 489.7 64 475.4 64 448V64c0-27.4 31.9-41.7 52.5-24.6L312 219.3V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="display: flex; align-items: center; justify-content: center; margin-top: 48px; font-size: 32px;"
    >
      <div>
        00:00
      </div>
      <input
        class="dark"
        max="0"
        min="0"
        style="width: 100%; margin: 0px 24px;"
        type="range"
        value="0"
      />
      <div>
        00:00
      </div>
    </div>
  </div>
</div>
`;

exports[`The MusicPlayer component renders with dark UI elements when the background is light 1`] = `
<div
  style="display: block; margin: 0px; position: absolute; height: 100%; width: 100%; color: black; font-size: 18px;"
>
  <div
    style="height: 5vh; width: 90vw; text-align: center; padding: 5vh 5vw 2vh 5vw;"
  >
    <div
      style="font-size: 24px;"
    >
      NOTHING PLAYING
    </div>
    <div
      style="font-size: 48px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
    >
      Please make a song selection
    </div>
  </div>
  <div
    class="album-art"
    style="width: 80vw; height: 80vw; position: absolute; top: 42vh; left: 50vw; transform: translate(-50%, -50%); background-color: rgb(192, 192, 192); box-shadow: 0px 0px 64px 4px rgba(0,0,0,0.2);"
  >
    <div
      style="font-size: 400px; position: absolute; top: 50%; left: 48%; transform: translate(-50%, -50%);"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-music fa-w-16 "
        data-icon="music"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    style="height: 10vh; width: 90vw; position: absolute; bottom: 0px; padding: 0px 5vw 5vh 5vw;"
  >
    <audio
      class="react-audio-player "
      id=""
      preload="metadata"
      src=""
      title=""
    >
      <p>
        Your browser does not support the 
        <code>
          audio
        </code>
         element.
      </p>
    </audio>
    <div
      style="display: flex; align-items: center; justify-content: center;"
    >
      <div
        style="font-size: 48px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-step-backward fa-w-14 "
          data-icon="step-backward"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 468V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v176.4l195.5-181C352.1 22.3 384 36.6 384 64v384c0 27.4-31.9 41.7-52.5 24.6L136 292.7V468c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        style="font-size: 96px; margin: 0px 96px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-play fa-w-14 "
          data-icon="play"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        style="font-size: 48px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-step-forward fa-w-14 "
          data-icon="step-forward"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 44v424c0 6.6-5.4 12-12 12h-48c-6.6 0-12-5.4-12-12V291.6l-195.5 181C95.9 489.7 64 475.4 64 448V64c0-27.4 31.9-41.7 52.5-24.6L312 219.3V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="display: flex; align-items: center; justify-content: center; margin-top: 48px; font-size: 32px;"
    >
      <div>
        00:00
      </div>
      <input
        class="dark"
        max="0"
        min="0"
        style="width: 100%; margin: 0px 24px;"
        type="range"
        value="0"
      />
      <div>
        00:00
      </div>
    </div>
  </div>
</div>
`;

exports[`The MusicPlayer component renders with light UI elements when the background is dark 1`] = `
<div
  style="display: block; margin: 0px; position: absolute; height: 100%; width: 100%; color: white; font-size: 18px;"
>
  <div
    style="height: 5vh; width: 90vw; text-align: center; padding: 5vh 5vw 2vh 5vw;"
  >
    <div
      style="font-size: 24px;"
    >
      NOTHING PLAYING
    </div>
    <div
      style="font-size: 48px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
    >
      Please make a song selection
    </div>
  </div>
  <div
    class="album-art"
    style="width: 80vw; height: 80vw; position: absolute; top: 42vh; left: 50vw; transform: translate(-50%, -50%); background-color: rgb(192, 192, 192); box-shadow: 0px 0px 64px 4px rgba(0,0,0,0.2);"
  >
    <div
      style="font-size: 400px; position: absolute; top: 50%; left: 48%; transform: translate(-50%, -50%);"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-music fa-w-16 "
        data-icon="music"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    style="height: 10vh; width: 90vw; position: absolute; bottom: 0px; padding: 0px 5vw 5vh 5vw;"
  >
    <audio
      class="react-audio-player "
      id=""
      preload="metadata"
      src=""
      title=""
    >
      <p>
        Your browser does not support the 
        <code>
          audio
        </code>
         element.
      </p>
    </audio>
    <div
      style="display: flex; align-items: center; justify-content: center;"
    >
      <div
        style="font-size: 48px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-step-backward fa-w-14 "
          data-icon="step-backward"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 468V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v176.4l195.5-181C352.1 22.3 384 36.6 384 64v384c0 27.4-31.9 41.7-52.5 24.6L136 292.7V468c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        style="font-size: 96px; margin: 0px 96px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-play fa-w-14 "
          data-icon="play"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        style="font-size: 48px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-step-forward fa-w-14 "
          data-icon="step-forward"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 44v424c0 6.6-5.4 12-12 12h-48c-6.6 0-12-5.4-12-12V291.6l-195.5 181C95.9 489.7 64 475.4 64 448V64c0-27.4 31.9-41.7 52.5-24.6L312 219.3V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="display: flex; align-items: center; justify-content: center; margin-top: 48px; font-size: 32px;"
    >
      <div>
        00:00
      </div>
      <input
        class="light"
        max="0"
        min="0"
        style="width: 100%; margin: 0px 24px;"
        type="range"
        value="0"
      />
      <div>
        00:00
      </div>
    </div>
  </div>
</div>
`;
